waspc/src/Wasp/Util.hs

```diff
@@ -102,3 +116,20 @@ concatShortPrefixAndText prefix text =
 concatPrefixAndText :: String -> String -> String
 concatPrefixAndText prefix text =
   if length (lines text) <= 1 then prefix ++ text else prefix ++ "\n" ++ indent 2 text
+
+type Checksum = Hex
+
+checksumFromString :: String -> Checksum
+checksumFromString = bytestringToHex . SHA256.hash . BSU.fromString
+
+checksumFromText :: Text -> Checksum
+checksumFromText = bytestringToHex . SHA256.hash . TextEncoding.encodeUtf8
+
+checksumFromByteString :: BSU.ByteString -> Checksum
+checksumFromByteString = bytestringToHex . SHA256.hash
+
+newtype Hex = Hex String
+  deriving (Show, Eq, Ord, Aeson.ToJSON, Aeson.FromJSON)
+
+bytestringToHex :: B.ByteString -> Hex
+bytestringToHex = Hex . concatMap (printf "%02x") . B.unpack
```

Comment:
Lifted all the above, and the tests, from our in-progress pair programming work.